File: ALODWebUtility/Common/PEPPType.cs
using System;
using System.Data;
using ALOD.Core.Domain.Lookup;
using ALOD.Data;

namespace ALODWebUtility.Common
{
    public class PEPPType : IntValuedLookupItem
    {
        protected int _active = 0;            // active

        #region Properties

        public int Active
        {
            get { return _active; }
            set { _active = value; }
        }

        #endregion

        // Get all PEPPType records from the lookup table for the Sys Admin
        public override DataTable GetRecords()
        {
            ILookupDao lkup = new NHibernateDaoFactory().GetLookupDao();
            return lkup.GetPEPPTypes(0, 0).Tables[0];
        }

        // Insert new PEPPType into the lookup table
        public void InsertPEPPType(int Value, string Name, int Active)
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPType", Value, Name, Active);
        }

        // Edit/Update an existing PEPPType in the lookup table
        public void UpdatePEPPType(int Value, string Name, int Active)
        {
            SqlDataStore adapter = new SqlDataStore();
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPType", Value, Name, Active);
        }
    }
}

File: ALODWebUtility/Common/PEPPType.vb
Imports ALOD.Core.Domain.Lookup
Imports ALOD.Data

Namespace Common

    Public Class PEPPType
        Inherits IntValuedLookupItem

        Protected _active As Integer = 0            ' active

#Region "Properties"

        Public Property Active() As Integer
            Get
                Return _active
            End Get

            Set(value As Integer)
                _active = value
            End Set
        End Property

#End Region

        ' Get all PEPPType records from the lookup table for the Sys Admin
        Public Overrides Function GetRecords() As DataTable
            Dim lkup As ILookupDao = New NHibernateDaoFactory().GetLookupDao()
            Return lkup.GetPEPPTypes(0, 0).Tables(0)
        End Function

        ' Insert new PEPPType into the lookup table
        Public Sub InsertPEPPType(ByVal Value As Integer, ByVal Name As String, ByVal Active As Integer)
            Dim adapter As New SqlDataStore()
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPType", Value, Name, Active)
        End Sub

        ' Edit/Update an existing PEPPType in the lookup table
        Public Sub UpdatePEPPType(ByVal Value As Integer, ByVal Name As String, ByVal Active As Integer)
            Dim adapter As New SqlDataStore()
            adapter.ExecuteNonQuery("core_lookups_sp_Insert_PEPPType", Value, Name, Active)
        End Sub

    End Class

End Namespace